aspnetcore/signalr/javascript-client.md

```diff
@@ -98,7 +98,140 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
 
 ## Reconnect clients
 
-The JavaScript client for SignalR doesn't automatically reconnect. You must write code that will reconnect your client manually. The following code demonstrates a typical reconnection approach:
+::: moniker range=">= aspnetcore-3.0"
+
+### Automatically reconnect
+
+The JavaScript client for SignalR can be configured to automatically reconnect using the [withAutomaticReconnect](/javascript/api/%40aspnet/signalr/hubconnectionbuilder#withautomaticreconnect) method on [HubConnectionBuilder](/javascript/api/%40aspnet/signalr/hubconnectionbuilder). It won't automatically reconnect by default.
+
+```javascript
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("/chatHub")
+    .withAutomaticReconnect()
+    .build();
+```
+
+Without any parameters, `withAutomaticReconnect()` configures the client to wait 0, 2, 10, and 30 seconds respectively before trying each reconnect attempt, stopping after four failed attempts.
+
+Before starting any reconnect attempts, the `HubConnection` will transition to the [Reconnecting](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onreconnecting](/javascript/api/%40aspnet/signalr/hubconnection#onreconnecting) callbacks instead of transitioning to the [Disconnected](javascript/api/%40aspnet/signalr/hubconnectionstate#disconnected) state and triggering its [onclosed](/javascript/api/%40aspnet/signalr/hubconnection#onclosed) callbacks like a `HubConnection` without automatic reconnect configured. This provides an opportunity to warn users that the connection has been lost and to disable UI elements.
+
+```javascript
+connection.onreconnecting((error) => {
+  console.assert(connection.state === signalR.HubConnectionState.Reconnecting);
+
+  document.getElementById("messageInput").disabled = true;
+
+  const li = document.createElement("li");
+  li.textContent = `Connection lost due to error "${error}". Reconnecting.`;
+  document.getElementById("messagesList").appendChild(li);
+});
+```
+
+If the client successfully reconnects within its first four attempts, the `HubConnection` will transition back to the [Connected](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onreconnected](/javascript/api/%40aspnet/signalr/hubconnection#onreconnected) callbacks. This provides an opportunity to inform users the connection has been reestablished.
+
+Since the connection looks entirely new to the server, a new `connectionId` will be provided to the `onreconnected` callback.
+
+> [!WARNING]
+> The `onreconnected` callback's `connectionId` parameter won't be undefined if the `HubConnection` was configured to [skip negotiation](xref:signalr/configuration#configure-client-options).
```

Comment:
"won't be undefined" you mean "will be"?

Reply:
```suggestion
> The `onreconnected` callback's `connectionId` parameter will be undefined if the `HubConnection` was configured to [skip negotiation](xref:signalr/configuration#configure-client-options).
```